File: algorithm/src/main/java/baekjoon/p01476/Main.java
package baekjoon.p01476;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt(), sc.nextInt(), sc.nextInt()));
    }

    private static int solve(int e, int s, int m) {
        int ret = 1;

        while (true) {
            int a = (ret - e) % 15;
            int b = (ret - s) % 28;
            int c = (ret - m) % 19;
            if (a == 0 && b == 0 && c == 0) {
                break;
            }
            ret++;
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p01476/Main.kt
package baekjoon.p01476

fun main(args: Array<String>) {
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    println(solve(inputs))
}

private fun solve(inputs: List<Int>): Int {
    var ret = 1
    val inputArr = inputs.toIntArray()
    val list = IntArray(3) { 1 }
    val set = intArrayOf(15, 28, 19)

    while (!inputArr.contentEquals(list)) {
        for (i in 0 until list.size) {
            list[i]++
            if (list[i] > set[i]) {
                list[i] = 1
            }
        }

        ret++
    }

    return ret
}